app/src/main/java/comp_sci_squad/com/github/url_irl/MainActivity.java

```diff
@@ -221,11 +95,52 @@ public void onClick(View v) {
             if (v.getId() == R.id.shutter_button) {
                 if (mCamera != null || mEmulatorPreview != null) {
                     mShutterButton.setEnabled(false);
-
-                    if (mEmulated)
+                    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+                    if (mEmulated) {
                         GlideApp.with(MainActivity.this).load(R.raw.tester_pic_remove_on_release).into(mCapturedImagePreview);
-                    else
-                        mCamera.takePicture();
+                    } else {
+                        // TODO: Offload most of this work off the MainThread asynchronously
+                        Log.d(TAG, "Picture taken");
+                        final long timeImageTaken = System.currentTimeMillis();
+
+                        Log.d(TAG, "Getting picture Dimensions");
+                        Bitmap bmp = mPreview.getBitmap();
+                        if (bmp == null) {
+                            Log.e(TAG, "Preview Image Bitmap is null");
+                            return;
+                        }
+                        bmp.compress(Bitmap.CompressFormat.PNG, 100, stream);
```

Comment:
Are you role-playing Snapchat by taking a screenshot of the image preview rather than actually taking a picture???

Reply:
Lmao, I can add a new bug to actually take a picture and load it in cache. We are essentially doing it that now.

Reply:
By doing that now, I meant we used to also just take a bitmap of the camera view itself and not the picture.